posthog/api/sharing.py

```diff
@@ -250,6 +260,25 @@ def retrieve(self, request: Request, *args: Any, **kwargs: Any) -> Any:
         }
         exported_data: dict[str, Any] = {"type": "embed" if embedded else "scene"}
 
+        if "whitelabel" in request.GET and "white_labelling" in [
+            feature["key"] for feature in resource.team.organization.available_product_features
+        ]:
+            exported_data.update({"whitelabel": True})
+
+        if isinstance(resource, SharingConfiguration) and resource.password_required:
+            if request.method == "GET":
+                exported_data["type"] = "login"
+                return render_template(
+                    "exporter.html",
+                    request=request,
+                    context={
+                        "exported_data": json.dumps(exported_data, cls=DjangoJSONEncoder),
+                        "add_og_tags": None,
+                    },
+                )
+            if "password" not in request.data or request.data["password"] != resource.password:
+                return response.Response({"error": "Incorrect password"}, status=401)
```

Comment:
FYI - this isn't actually all that secure. The `access_token` (which is in the URL) is passed to all API requests and validated with [this permission](https://github.com/PostHog/posthog/blob/master/posthog/permissions.py#L245). So the password stops you from accessing the initial webpage, but it doesn't stop you from accessing the API calls that hold the data.

Either we should document that this is not truly a secure method and is just an extra layer of obscurity or you need to consider how to handle that other case. Something like:
1. Validate the password on post
2. If valid generate a specific JWT (see existing helpers) that has the AUD scoped only to the sharing token
3. Modify the permission checker to ensure that it ignores access_tokens that have password_required
4. Modfiy it also then to validate the JWT instead in that case.